src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -0,0 +1,194 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { repeat } from "lit-html/directives/repeat";
+
+import "../../../components/ha-card.js";
+import "../components/hui-entities-toggle.js";
+
+import { fireEvent } from "../../../common/dom/fire_event.js";
+import { DOMAINS_HIDE_MORE_INFO } from "../../../common/const.js";
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { HomeAssistant } from "../../../types.js";
+import createRowElement from "../common/create-row-element.js";
+import computeDomain from "../../../common/entity/compute_domain.js";
+import processConfigEntities from "../common/process-config-entities";
+
+interface EntityConfig {
+  name?: string;
+  icon?: string;
+  entity: string;
+  type?: string;
+  secondary_info: "entity-id" | "last-changed";
+  action_name?: string;
+  service?: string;
+  service_data?: object;
+  url?: string;
+}
+
+interface Config extends LovelaceConfig {
+  show_header_toggle?: boolean;
+  title?: string;
+  entities: EntityConfig[];
+}
+
+class HuiEntitiesCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+  protected configEntities?: EntityConfig[];
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    // +1 for the header
+    return 1 + (this.config ? this.config.entities.length : 0);
+  }
+
+  public setConfig(config: Config) {
+    this.config = { show_header_toggle: true, ...config };
+    const entities = processConfigEntities(config.entities);
+
+    for (const entity of entities) {
+      if (
+        entity.type === "call-service" &&
+        !entity.service &&
+        !entity.name &&
+        !entity.icon &&
+        !entity.service_data &&
+        !entity.action_name
+      ) {
+        throw new Error("Missing required property when type is call-service");
+      } else if (
+        entity.type === "weblink" &&
+        !entity.name &&
+        !entity.icon &&
+        !entity.url
+      ) {
+        throw new Error("Missing required property when type is weblink");
+      }
+    }
+
+    this.configEntities = entities;
+    if (this.hass) {
+      this.requestUpdate();
+    }
+  }
+
+  protected render() {
+    if (!this.config || !this.hass) {
+      return html``;
+    }
+    const { show_header_toggle, title } = this.config;
+    const states = this.hass.states;
+
+    const entities = this.configEntities!.filter(
+      (conf) => conf.entity in states
+    );
+
+    const toggableEntities = entities
+      .filter((item) => typeof item === "string" || item.entity)
```

Comment:
I don't ttink we need to check for string because processConfigEntities converts it all to objects